core/app/models/uffizzi_core/compose_file.rb

```diff
@@ -54,4 +53,12 @@ class UffizziCore::ComposeFile < UffizziCore::ApplicationRecord
       transitions from: [:valid_file], to: :invalid_file
     end
   end
+
+  private
+
+  def main_compose_not_exists
+    return unless project.compose_files.any? { |compose_file| compose_file.kind.main? }
```

Comment:
Let's use a scope:
```
return unless project.compose_files.main.exists?
```

Or `with_kind(:main)`, I don't remember which is correct